pkg/jobs/hive/hive.go

```diff
@@ -329,8 +333,268 @@ func UpgradeCluster(client clientv1.Client, clusterName string, curator *cluster
+		"-kube-1." + strconv.Itoa(int(desiredSemVer.Minor+14)) + "-api-removals-in-" +
+		strconv.Itoa(int(desiredSemVer.Major)) + "." + strconv.Itoa(int(desiredSemVer.Minor+1))
+
+	cmData := configMap["data"]
+	if cmData != nil {
+		// "kube-1.{{ '%02d' | format((ocp_minor_version.split('.')[1] | int) + 14"
+		cmData.(map[string]interface{})[ackString] = "true"
+	} else {
+		configMap["data"] = map[string]interface{}{
+			ackString: "true",
+		}
+	}
+
+	// update managed cluster with acknowledge config to allow upgrade
+	var updateConfigMap runtime.RawExtension
+	if configMap != nil {
+		b, err := json.Marshal(configMap)
+		utils.CheckError(err)
+		updateConfigMap.Raw = b
+	}
+	klog.V(2).Info("Create managedclusteraction to update configmap " + clusterName + "admack")
+	ocpConfigMCA := &managedclusteractionv1beta1.ManagedClusterAction{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      clusterName + "admack",
+			Namespace: clusterName,
+		},
+		Spec: managedclusteractionv1beta1.ActionSpec{
+			ActionType: managedclusteractionv1beta1.UpdateActionType,
+			KubeWork: &managedclusteractionv1beta1.KubeWorkSpec{
+				Resource:       "configmaps",
+				Name:           "admin-acks",
+				Namespace:      "openshift-config",
+				ObjectTemplate: updateConfigMap,
+			},
+		},
+	}
+	if err := client.Create(context.TODO(), ocpConfigMCA); err != nil {
+		return err
+	}
+
+	// wait for managedclusteraction results
+	ocpConfigMCAStatus := managedclusteractionv1beta1.ManagedClusterAction{}
+	for i := 1; i <= 5; i++ {
+		time.Sleep(utils.PauseFiveSeconds)
+		if err := client.Get(context.TODO(), types.NamespacedName{
+			Name:      clusterName + "admack",
+			Namespace: clusterName,
+		}, &ocpConfigMCAStatus); err != nil {
+			if i == 5 {
+				return err
+			}
+			klog.Warning(err)
+			continue
+		}
+		if ocpConfigMCAStatus.Status.Conditions != nil {
+			break
+		}
+	}
+
+	if ocpConfigMCAStatus.Status.Conditions != nil {
+		for _, condition := range ocpConfigMCAStatus.Status.Conditions {
+			if condition.Status == v1.ConditionFalse && condition.Reason == managedclusteractionv1beta1.ReasonUpdateResourceFailed {
+				klog.Warning("ManagedClusterAction failed to update remote clusterversion", ocpConfigMCAStatus.Status.Conditions)
+				return errors.New("Remote confimap update failed")
+			}
+			if condition.Status == v1.ConditionTrue && condition.Type == managedclusteractionv1beta1.ConditionActionCompleted {
+				klog.V(2).Info("Remote configmap updated successfully " + clusterName + "admack")
+			}
+		}
+	} else {
+		return errors.New("Remote configmap update failed")
+	}
+
+	if err := client.Delete(context.TODO(), &ocpConfigMCAStatus); err != nil {
+		return err
+	}
+
+	// After updating the OCP ack configmap it takes some time for clusterversion to pick up the change
+	klog.V(0).Info("Pause 60 seconds for clusterversion to update")
+	time.Sleep(utils.PauseSixtySeconds)
```

Comment:
do we know what we are looking for so we know clusterversion is done updating? 
should we do another mcv or mca here to check the status instead of hoping the 60s is enough?

Reply:
So I added this before I added the retry for updating the clusterversion. There's message in the OCP console saying the admin must acknowledge the removed Kube APIs which should disappear. I'm not sure how easy it is to check for that and maybe this is not needed anymore since we have the retry in place?